### DISCUSSION TITLE
윤태연

### DISCUSSION THREAD

**Discussion by z1nun** [z1nun/ASTERISK-#14] [독서일지]
# 각자 선정한 개발자 필독서를 읽고 그날 읽은 내용을 간단히 정리해서 적어주세요!

## 오늘도 개발자가 안 된다고 말했다
> 내용정리 시작! 한 Part당 소제목 1개씩 읽을 것이다!

### PART 01 가깝고도 먼 개발자
1. 어딘가 이상한 비전공자의 협업 : 22/10/03
2. 온몸으로 느낀 개발자 : 22/10/06
3. 협업을 위한 준비물
### PART 02 기획자의 일
1. 서비스 기획 들여다보기
2. 협업을 위한 사전 준비
3. 협업을 돕는 화면 설계서
### PART 03 디자이너의 일
1. 디자이너의 마인드셋
2. 정확한 시각화를 위한 개발 지식
3. 협업을 위한 개발 지식
### PART 04 개발자의 일
1. 개발자 이해하기
2. 생산성 향상을 위한 협업 툴
3. 개발자가 말하는 협업

**Comment by rhino-ty** [comment]
2022 / 10 / 03
# Part1. 가깝고도 먼 개발자
## 1. 어딘가 이상한 비전공자의 협업
IT업계 회사엔 개발, 기획, 디자인 등 다양한 인력들이 모여 협업을 이룬다. 그 중 개발자와의 협업을 제일 어려워 하는 이야기가 많이들 나온다. 왜 이런 것일까? 그 답은 개발 언어가 거의 외국어 수준이여서 소통이 안되는 점, 개발이 어떻게 되는지에 대한 과정 및 구조를 전혀 알 수 없어서 나온다. 결국 기획하고자 하는 의도, 웹 구조에 의한 웹 디자인 등 기본적인 개발을 알아야 원활한 소통이 가능하다. 즉 협업의 기본은 최소한 개발에 대해 과정과 구조를 겉으로도, 피상적인 이해가 있어야 협업이 가능하다. 그러므로 자신의 회사에 필요한 기술 스택을 알고, **협업에 필요한 개발 지식을 학습하여 원활한 협업을 요하는 것이 중요**하다!

# _내용 정리_
# Part1. 가깝고도 먼 개발자
## 1. 어딘가 이상한 비전공자의 협업
### - 기획자 김 군의 협업
행정학과 김 군, 교내 스타트업 스타팅 멤버로 PT 할 수 있다는 이유로 취업, 서투른 기획서로 서비스 기획을 하며 개발자와 협업, 규모가 작아 서로의 의도를 잘 파악할 수 있는 구조라 원만한 협업 가능, 협업 중 기획하고자 하는 의도, 즉 '메시지'가 중요하다는 걸 김 군은 깨달음

하지만 클라이언트(고객)와도 소통이 필요, 만일 WIFI를 사용하는데 광고를 봐야 사용 가능케 하는 서비스를 제작해 팔아 지점마다 설치하는 서비스를 김 군이 기획, 영업, 유지•관리함, 그런데 만일 클라이언트에 문제가 생기거나 해주지 못할 요청사항 등 김 군이 커트해야할 문제들을 관리해야하는데 화면 관리의 프론트엔드, 서버의 백엔드, WIFI의 펌웨어, 네트워크 등 다양한 기술 문제를 모른다면 관리하지 못함

또한 개발자들끼리의 생각이 달라 기획자로서 조율해아하는 문제에 부닥칠 수도 있음, '프엔 개발자는 된다지만 백엔 개발자는 기능 구현이 안된다.' 식의 문제에 직면할 수도 있음, 하지만 백엔드 개발자가 기능 성장을 저해한다든지 그런 의미가 아님, 서비스 운용의 관점이 다르거나 어떠한 문제로 부딪혀 부정적인 생각을 내비치는 것일 수도 있음

즉! 기획자는 **개발 구조를 이해해 전체적인 개발 흐름을 알고, 개발자와의 소통을 하는데 있어 문제의 대안이나 더 좋은 서비스로 성장하자는 의도 등 유연한 사고방식을 가져 개발자와 보다 더 나은 파트너의 관계로 발전해 나갈 수 있다.**
### - 디자이너 김 양의 협업
중장기 계획을 짜고 사업의 전체적인 전략을 짜는 소규모 IT기업의 전략기획부로 입사한 김 양, 하지만 김 양은 디자인 툴과 블로그 경험이 있어 입사한 터라 간단한 디자인 업무와 마케팅 업무를 도맡아 했음, 그러다 대행 판매 건을 맡게 됐고 페이지가 필요해 개발이 필요한 '랜딩 페이지'를 디자인해 개발팀에게 전달했고 당연히 반려당함 - 웹 개발에 기반하지않은 디자인 기획(구역, 모션 등), 프로젝트가 어떻게 진행되는지 파악 어려움

그러면서 깨닫게 됨
   1. 웹 디자인은 웹이 동작하는 원리를 알고 그에 맞는 디자인 기획을 짜야함
   2. 기본적인 개발 구조 파악 : 프엔, 백엔, DB 등 개발자들이 하는 일이 다 다르다는 걸 인식하게 됨

이러한 점을 깨달으며 프로젝트의 어떤 기능에 대해 맞는 개발자와 얘기해야할지, 일정 내 구현이 가능한지에 대해 생각하게 됨

결국 웹 개발에 대해 다시 알아야하는 김 양은 HTML, CSS, JS를 공부하며 기초를 다졌고, 자신이 배운 데까지만 기능과 디자인 구현을 하게 됨. '프엔 개발자가 되볼까?' 하는 생각도 들었지만 결국 디자이너로서 가지는 디자인 관점을 포기하고 코드로만 생각하게 되는 자신을 보고 깨달음. 개발 언어, 프로그램 개발 자체가 아니라 나의 회사에서 개발되는 구조, 과정 등을 알고 나의 디자인이 어떻게 구축되는지만 알면 이 관점을 유지하면서 원만히 협업할 수 있구나~하고.

결국 디자이너는 사용자에게 어떻게 보이는지, 어떠한 환경에서 어떻게 관리하며 경험 상 어떻게 사용할지 등을 알아 내가 보는 게 아닌 사용자가 실제로 접근하는 것이 최종완성본이란 걸 아는 게 중요함. 그래서 그 이전 개발에 대해 이해하여 개발자와의 소통을 원활히 하면서 디자이너로서 이런 부분을 함께 체크한다면 좋은 협업이 가능할 것임
### - 우리가 협업을 할 수 있었던 이유
위 김 군이 처음 협업할 시 협업을 가능케 했던 이유는 '하얀색 도화지'였고 소규모였음, 그래서 서로의 포지션에서 서로의 말을 경청하고 의도를 보다 쉽게 파악할 수 있어 이해 되지 않은 내용은 반복적인 질문, 필요한 정보를 파악 후 빠른 대처가 가능했었음. 즉, 협업의 기본을 다 할 수 있어 원만한 소통이 가능했음. 
하지만 일반적인 회사에서 근속하며 그러기엔 쉽지 않음. 그러므로 개발자에게 질문을 해 협업에 필요한 개발 지식 습득 등 자신의 회사에 필요한 개발 기술 지식이 무엇인지 빠르게 알아 개발자와의 원만한 협업을 위해 노력하자

**Comment by rhino-ty** [comment]
2022 / 10 / 06
# Part1. 가깝고도 먼 개발자
## 2. 온몸으로 느낀 개발자
여기선 개발자가 아닌 포지션이 개발자를 관찰하여 알게된 것을 사례로 3가지를 들며 서술하고있다. 
1. 자주 부정적 개발자(상황을 고려해 대안 제시)
2. 매우 긍정적 개발자(오히려 무서움, 확실한 조건 걸기) 
3. 대안 제시 개발자(상황 인지해 대안 제시)

다양한 유형이 있지만 쨌든 좋은 개발자는 '회사의 성장을 위해 협업해 문제를 해결하는 개발자'라는 것이다. 그런 개발자의 특징을 4가지로 나눈다.
1. 집요한 문제 해결(문제 해결에 집요, 해결함으로써 즐거움을 느낌) 
2. 비즈니스 이해하는 눈(고객 관점 중시, 비즈니스 모델 추구 및 빠른 테스트) 
3. 5살배기도 이해하는 소통의 기술(어려운 개발 용어를 쉽게 풀어서) 4. 체계적 업무 관리와 빠른 피드백(업무 일정 내 마감, 변수에 빠른 피드백)

또한 다른 포지션들에게도 3가지 교훈을 던져준다. 개발자들이 그냥 말에도 그렇게 '느낄 수 있는' 포인트를 던져준다.
1. 간단하죠? 해주세요(상대방 역할 무시 어투, 하나 버튼 추가에 복잡한 기능 연결) 
2. ~까지 해주세요(개발자 입장 고려 X)
3. 타 서비스는 제공하던데요?(타 개발자와 비교 어투, 안된다는 것엔 여러 이유 존재)
